dask_match/expr.py

```diff
@@ -887,6 +907,81 @@ def simplify(self):
                 for op in self.frame.operands
             ]
             return type(self.frame)(*operands)
+        elif isinstance(self.frame, PartitionsFiltered):
+            if self.frame._partitions:
+                partitions = [self.frame._partitions[p] for p in self.partitions]
+            else:
+                partitions = self.partitions
+            # We assume that expressions defining a special "_partitions"
+            # parameter can internally capture the same logic as `Partitions`
+            operands = [
+                partitions if self.frame._parameters[i] == "_partitions" else op
+                for i, op in enumerate(self.frame.operands)
+            ]
```

Comment:
We do this pattern a lot. I want to think about some dict variant of operands in the future I think

Reply:
I was thinking about that as well. One complication is that we don't want to require a key for every operand (recall that variadic expressions are useful). Therefore, it may be less restrictive if we just support something like `Expr.replace_operands(**operand_replacements)` so that this logic can be isolated in one place.